CMakeLists.txt

```diff
@@ -727,6 +727,11 @@ cmake_dependent_option(
   "Whether to build the flash_attention kernel for scaled dot product attention" ON
   "USE_CUDA AND NOT ROCM AND NOT CMAKE_CUDA_COMPILER_VERSION VERSION_LESS 11.6" OFF)
 
+if(USE_FLASH_ATTENTION)
+  message("USE_FLASH_ATTENTION SET, setting max jobs to 1")
+  set(ENV{MAX_JOBS} "1")
```

Comment:
This approach looks right to me if the OOM issue is caused by cmake uses too many parallel processes and doesn't limit itself to the amount of available memory.  Is there a way to set this to MAX_JOBS / 2 or MAX_JOBS - 4 here, same as what we are using to build libtorch https://github.com/pytorch/pytorch/blob/main/.ci/pytorch/build.sh#L301-L305?